DigitalLearningSolutions.Web/Views/MyAccount/_EditRegistrationNumber.cshtml

```diff
@@ -3,27 +3,28 @@
 @model EditDetailsViewModel
 
 @{
-  var errorHasOccurred = !ViewData.ModelState.IsValid && @Model.ProfessionalRegNumberSelectionAnswer == YesNoSelectionEnum.None;
-  var optionYesSelected = @Model.ProfessionalRegNumberSelectionAnswer == YesNoSelectionEnum.Yes;
-  var optionNoSelected = @Model.ProfessionalRegNumberSelectionAnswer == YesNoSelectionEnum.No;
+  var errorHasOccurred = !ViewData.ModelState.IsValid;
+  var optionNoneSelected =  @Model.HasProfessionalRegistrationNumber == YesNoSelectionEnum.None;
+  var optionYesSelected = @Model.HasProfessionalRegistrationNumber == YesNoSelectionEnum.Yes;
+  var optionNoSelected = @Model.HasProfessionalRegistrationNumber == YesNoSelectionEnum.No;
   var professionalRegConditionalId = "professional-reg-conditional-Id";
 }
 
-<div class="nhsuk-form-group @(errorHasOccurred? "nhsuk-form-group--error": "")">
+<div class="nhsuk-form-group @(optionNoneSelected? "nhsuk-form-group--error": "")">
   <fieldset class="nhsuk-fieldset" aria-describedby="professional-registration-hint">
     <div class="nhsuk-hint" id="professional-registration-hint">
       Do you have a Professional Registration Number?
     </div>
 
-  @if (errorHasOccurred) {
+  @if (errorHasOccurred && optionNoneSelected) {
     <span class="nhsuk-error-message" id="no-selection-error">
-      <span class="nhsuk-u-visually-hidden">Error:</span> @(!optionYesSelected? "Select an option":"")
+       @(!optionYesSelected? "Select an option":"")
     </span>
   }

```

Comment:
We definitely still want the visually hidden `Error: ` so that a screen reader announces this as being an error to a user. What we don't want is to have the logic of whether and error has occurred and what the error message should be to be set in the view - that's the responsibility of the model.
We should be displaying an error here only if there is an error on the `HasProfessionalRegistrationNumber` property and then we should be displaying the error message from that error, not hard-coding a string in the view. You can check e.g. `TextInputViewComponent.cs` to see how to get the error messages for a specific property.

Reply:
I have done this change